### DISCUSSION TITLE
Error Creating S-Function

### DISCUSSION THREAD

**Discussion by liruiqi97** [OpenFAST/openfast#1137] [Q&A] [answered]
Dear developers,

Thanks for your patience! I am trying to make some wind-turbine simulations using OpenFAST with MATLAB Simulink, but get confused when running the glue-codes.

I am using Ubuntu 20.04 system which is Linux. Now I have downloaded the Github source codes and installed using "cmake" and "make" on terminal. However, I encountered errors when running `openfast/glue-codes/simulink/src/create_FAST_SFunc.m`.

![Screenshot from 2022-05-24 17-37-33](https://user-images.githubusercontent.com/43281955/170137897-58d1d8cf-edd3-4175-96ab-42b1d40a3ce0.png)

Something confusing me is that, the MATLAB files (e.g. `create_FAST_SFunc.m`) suppose that the install directory is `openfast/install`, but in the `openfast/build/CMakeCache.txt` the default install prefix is another directory `CMAKE_INSTALL_PREFIX:PATH=/usr/local` which is where the install files are actually created. Should I change this default setting? What I did is just copying the `bin`, `lib`, `include` folders from `/usr/local` into `openfast/install`, but I still encounter the issue above.

All advice is appreciated!

**Comment by bjonkman** [comment, marked as answer]
If the cmake install directory is specified differently than what is specified in create_FAST_SFunc, you should change the variable in create_FAST_SFunc.m to the value that cmake used (or you could also change the cmake install path before building OpenFAST for Simulink). Copying the directory should do the same thing, though.

This looks like a linking problem. It seems like it's not finding some of the gfortran routines. Is Matlab building this mex function with the same compiler (and same version) that you used with cmake to build the OpenFAST library for Simulink?
